pkgs/build-support/node/build-npm-package/npm.patch

```diff
@@ -0,0 +1,83 @@
+diff --git a/lib/node_modules/npm/node_modules/@npmcli/run-script/lib/set-path.js b/lib/node_modules/npm/node_modules/@npmcli/run-script/lib/set-path.js
+index c59c270d9..34f59a623 100644
+--- a/lib/node_modules/npm/node_modules/@npmcli/run-script/lib/set-path.js
++++ b/lib/node_modules/npm/node_modules/@npmcli/run-script/lib/set-path.js
+@@ -12,7 +12,7 @@ const setPATH = (projectPath, binPaths, env) => {
+     .reduce((set, p) => set.concat(p.filter(concatted => !set.includes(concatted))), [])
+     .join(delimiter)
+ 
+-  const pathArr = []
++  const pathArr = [nodeGypPath, PATH]
+   if (binPaths) {
+     pathArr.push(...binPaths)
+   }
+@@ -26,7 +26,6 @@ const setPATH = (projectPath, binPaths, env) => {
+     pp = p
+     p = dirname(p)
+   } while (p !== pp)
+-  pathArr.push(nodeGypPath, PATH)
+ 
+   const pathVal = pathArr.join(delimiter)
+ 
+diff --git a/lib/node_modules/npm/node_modules/pacote/lib/git.js b/lib/node_modules/npm/node_modules/pacote/lib/git.js
+index c4819b4fd..cc19af58c 100644
+--- a/lib/node_modules/npm/node_modules/pacote/lib/git.js
++++ b/lib/node_modules/npm/node_modules/pacote/lib/git.js
+@@ -186,15 +186,28 @@ class GitFetcher extends Fetcher {
+       }
+       noPrepare.push(this.resolved)
+ 
++      const spawn = require('@npmcli/promise-spawn')
++
++      const patchShebangs = () => spawn("@bash@", ["-c", "source @stdenv@/setup; patchShebangs node_modules"], { cwd: dir });
++
+       // the DirFetcher will do its own preparation to run the prepare scripts
+       // All we have to do is put the deps in place so that it can succeed.
+       return npm(
+         this.npmBin,
+-        [].concat(this.npmInstallCmd).concat(this.npmCliConfig),
++        ["ci", "--ignore-scripts"],
```

Comment:
Maybe add a comment for why this ignores `this.npmCliConfig`, and add in `npmFlags` and `npmInstallFlags` so that they get propogated for all recursive `npm` calls